lib/services/http-service.js

```diff
@@ -145,26 +146,43 @@ function httpServiceFactory(
     HttpService.prototype.createSwagger = function() {
         var app = this.app;
         var endpoint = this.endpoint;
+
+        function loadYamlConfig(yamlPath) {
+            var config = yaml.safeLoad(fs.readFileSync(yamlPath, 'utf8'));
```

Comment:
Don't have to change to the asynchronous version could add a try catch that returns a rejected promise.

```
  try {
    // ...
  } catch (err) {
    return Promise.reject(err)
  }
```